app/components/global-search.js

```diff
@@ -1,82 +1,78 @@
 import Ember from 'ember';
 
-const { Component, inject, run, computed } = Ember;
+const { Component, computed, inject, isEmpty, run } = Ember;
 const { service } = inject;
 const { debounce } = run;
-const { or, notEmpty } = computed;
+const { trim } = Ember.$;
+const { PromiseObject } = DS;
+
+function cleanQuery(query) {
```

Comment:
Splitting this up here is cleaner, but since it is only used once it should just go in the revisedQuery CP.  Otherwise you have to go looking for it when checking to see what it does.

We're not using this anywhere else, is it necessary?  Or just something you added for good measure?